packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -816,6 +816,81 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
         }
     }
 
+    /**
+     * @inheritdoc IAssetManager
+     */
+    function claimKghReward(address validator, uint256[] calldata tokenIds) external {
+        if (validator == address(0)) revert ZeroAddress();
+        if (tokenIds.length == 0) revert NotAllowedZeroInput();
+
+        Vault storage vault = _vaults[validator];
+        KghDelegator storage kghDelegator = vault.kghDelegators[msg.sender];
+        if (kghDelegator.kghNum < tokenIds.length) revert InvalidTokenIdsInput();
+
+        // Auto compounding is not applied for claim function.
+        uint128 claimedRewards = _claimBoostedReward(validator, msg.sender, tokenIds.length, false);
+
+        // Burn kroShares to claim the base reward.
+        uint128 kroSharesToBurn;
+        for (uint256 i = 0; i < tokenIds.length; ) {
+            uint128 kroShares = kghDelegator.shares[tokenIds[i]].kro -
+                _convertToKroShares(validator, KGH_MANAGER.totalKroInKgh(tokenIds[i]));
+            if (kroShares == 0) revert ShareNotExists();
```

Comment:
`kroShares` can be 0 if the delegator does not have any claimable KRO in KGH.

Reply:
That's valid. Thanks for pointing it out!